File: testdata/repos/commons/templates/create_formula/languages/java11/helloWorld/src/test/java/com/ritchie/formula/FormulaTest.java
package com.ritchie.formula;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;

import java.io.ByteArrayOutputStream;
import java.io.PrintStream;

import static org.junit.Assert.assertEquals;

public class FormulaTest {

  private final ByteArrayOutputStream outContent = new ByteArrayOutputStream();
  private final PrintStream originalOut = System.out;

  @Before
  public void setUpStreams() {
    System.setOut(new PrintStream(outContent));
  }

  @After
  public void restoreStreams() {
    System.setOut(originalOut);
  }

  @Test
  public void runTrueInput() {
    new Formula("Hello", true, "world", "pass").Run();

    assertEquals("Hello World!" +
            "My name is Hello." +
            "I've already created formulas using Ritchie." +
            "Today, I want to automate world." +
            "My secret is pass.", outContent.toString().replaceAll("\\r|\\n", "").replaceAll("\u001B\\[[;\\d]*m", ""));
  }

  @Test
  public void runFalseInput() {
    new Formula("Hello", false, "world", "pass").Run();

    assertEquals("Hello World!" +
            "My name is Hello." +
            "I'm excited in creating new formulas using Ritchie." +
            "Today, I want to automate world." +
            "My secret is pass.", outContent.toString().replaceAll("\\r|\\n", "").replaceAll("\u001B\\[[;\\d]*m", ""));
  }

  @Test
  public void runNoSecretsInput() {
    new Formula("Hello", false, "world", "").Run();

    assertEquals("Hello World!" +
            "My name is Hello." +
            "I'm excited in creating new formulas using Ritchie." +
            "Today, I want to automate world." +
            "My secret is .", outContent.toString().replaceAll("\\r|\\n", "").replaceAll("\u001B\\[[;\\d]*m", ""));
  }
}

File: testdata/repos/commons/templates/create_formula/languages/kotlin-jdk8/helloWorld/src/test/kotlin/com/ritchie/formula/FormulaTest.kt
package com.ritchie.formula

import org.junit.After
import org.junit.Before
import org.junit.Test
import java.io.ByteArrayOutputStream
import java.io.PrintStream
import kotlin.test.assertEquals

class FormulaTest {
    private val outContent: ByteArrayOutputStream = ByteArrayOutputStream()
    private val originalOut: PrintStream = System.out
    @Before
    fun setUpStreams() {
        System.setOut(PrintStream(outContent))
    }

    @After
    fun restoreStreams() {
        System.setOut(originalOut)
    }

    @Test
    fun runTrueInput() {
        Formula("Hello", true, "world", "pass").Run()
        assertEquals("Hello World!" +
                "My name is Hello." +
                "I've already created formulas using Ritchie." +
                "Today, I want to automate world." +
                "My secret is pass.", outContent.toString().replace(Regex("\\r|\\n"), "").replace(Regex("\u001B\\[[;\\d]*m"), ""))
    }

    @Test
    fun runFalseInput() {
        Formula("Hello", false, "world", "pass").Run()
        assertEquals("Hello World!" +
                "My name is Hello." +
                "I'm excited in creating new formulas using Ritchie." +
                "Today, I want to automate world." +
                "My secret is pass.", outContent.toString().replace(Regex("\\r|\\n"), "").replace(Regex("\u001B\\[[;\\d]*m"), ""))
    }

    @Test
    fun runNoSecretsInput() {
        Formula("Hello", false, "world", "").Run()
        assertEquals("Hello World!" +
                "My name is Hello." +
                "I'm excited in creating new formulas using Ritchie." +
                "Today, I want to automate world." +
                "My secret is .", outContent.toString().replace(Regex("\\r|\\n"), "").replace(Regex("\u001B\\[[;\\d]*m"), ""))
    }
}
